### PR TITLE
build: change bundling strategy to transform ES Module to CommonJS

### PR DESCRIPTION
**Pull request by zhenwenc**
We re-bundle the library that transforms the source code from ES Module to CommonJS due to Jest doesn't support ES Module property.

Details can be found here: https://github.com/digitalbazaar/cborld/pull/49

---

This PR changes the bundling strategy from using TypeScript to Rollup, where the transformation output will be combined into one file (`dist/main.js`) rather than multiple files in the `dist` folder. 

This is under the consideration of the NPM dependency of this library will be pulled from this git repository directly ([npm-install docs](https://docs.npmjs.com/cli/v7/commands/npm-install)):

```sh
npm install git+https://github.com/mattrglobal/digitalbazaar-cborld.git#{{ commit-hash }}
```

It requires the `dist` folder to be committed and pushed, which produces quite some noise when reviewing PRs.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -67,3 +67,18 @@ jobs:
...
+    steps:
+    - uses: actions/checkout@v2
+    - name: Use Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@v1
+    - run: npm install
+    - run: npm run build
+    - name: Check bundle output
+      run: |
+        if  [[ "$(git ls-files -m)" != "" ]]; then
+          echo "::error::Please build and commit these files:"
```
Comment (ivy-wei-yang) [inline .github/workflows/main.yml:81]: Check if the bundle file is committed or not 👍

**Review by ivy-wei-yang — DISMISSED** [review]

**Comment by zhenwenc** [comment]
NOTE: Just pushed `index.d.ts`. Unfortunately, we have to provide the type definition if we don't transpile the source code with `tsc`. 😞

**Review by ivy-wei-yang — DISMISSED** [review]

**Comment by aliXsed** [comment]
There seems to be alternative solutions to make sure npm would get the dist files without us pushing the dist folder to the git repo. Can you please have a look at https://stackoverflow.com/questions/31642477/how-to-publish-an-npm-package-with-distribution-files and see if those could be a better approach? I'm a bit wary of introducing `rollup`.

**Comment by zhenwenc** [comment]
> There seems to be alternative solutions to make sure npm would get the dist files without us pushing the dist folder to the git repo. Can you please have a look at https://stackoverflow.com/questions/31642477/how-to-publish-an-npm-package-with-distribution-files and see if those could be a better approach? I'm a bit wary of introducing `rollup`.

@aliXsed Thanks a lot for the suggestion 🙏 Do you mean by using the NPM `prepublish` lifecycle hook? If so, the situation is slightly different here, we don't actually publish the package, but reference to the git repository directly: https://docs.npmjs.com/cli/v7/commands/npm-install

**Comment by aliXsed** [comment]
> > There seems to be alternative solutions to make sure npm would get the dist files without us pushing the dist folder to the git repo. Can you please have a look at https://stackoverflow.com/questions/31642477/how-to-publish-an-npm-package-with-distribution-files and see if those could be a better approach? I'm a bit wary of introducing `rollup`.
> 
> @aliXsed Thanks a lot for the suggestion 🙏 Do you mean by using the NPM `prepublish` lifecycle hook? If so, the situation is slightly different here, we don't actually publish the package, but reference to the git repository directly: https://docs.npmjs.com/cli/v7/commands/npm-install

Ah Ok, now it's clearer to me. I think we should find a way to avoid pushing generated files to the repo. Even in the case of using "rollup", we are just combining them into one single file and pushing them. Here as you pointed out correctly, we don't publish to npm, so we need to find a different solution to have the dist files through "npm install". Can we use the "prepare" script for this purpose?

**Comment by zhenwenc** [comment]
> > > There seems to be alternative solutions to make sure npm would get the dist files without us pushing the dist folder to the git repo. Can you please have a look at https://stackoverflow.com/questions/31642477/how-to-publish-an-npm-package-with-distribution-files and see if those could be a better approach? I'm a bit wary of introducing `rollup`.
> > 
> > 
> > @aliXsed Thanks a lot for the suggestion 🙏 Do you mean by using the NPM `prepublish` lifecycle hook? If so, the situation is slightly different here, we don't actually publish the package, but reference to the git repository directly: https://docs.npmjs.com/cli/v7/commands/npm-install
> 
> Ah Ok, now it's clearer to me. I think we should find a way to avoid pushing generated files to the repo. Even in the case of using "rollup", we are just putting combining them into one file and push them. But here as don't publish to npm, we need to find a different solution to have the dist files through "npm install". Can we use the "prepare" script for this purpose?

The bundle step with "rollup" doesn't only concatenate the source files, but the main purpose is to transform source code written in ES Module (ESM) to CommonJS (CJS) format. You may find the reason here: https://github.com/digitalbazaar/cborld/pull/49

**Review by aliXsed — DISMISSED** [review]

**Review by ivy-wei-yang — APPROVED** [review]

**Review by aliXsed — APPROVED** [review]

**Pull request merged**